### ISSUE TITLE
Issue with controller count and node deletion

### ISSUE THREAD

**Issue by sanjaychari** [redhat-performance/jetpack#478]
I am trying to deploy an OSP 16.2 environment on scale lab with 2 different types of hardware(1029u and 1029p) with 16 nodes for the overcloud. One of the baremetal nodes got deleted as introspection failed. However, the compute count was not adjusted. OSP still provisioned 13 compute nodes, which left only 2 baremetal nodes for the controller. Because of this, the 3rd controller could not be allocated and failed with the error message below.

```
(undercloud) [stack@undercloud ~]$ openstack server show controller-0 --fit-width | grep fault
| fault                               | {'code': 500, 'created': '2022-09-20T10:58:39Z', 'message': 'No valid host was found. ', 'details': 'Traceback (most recent call last):\n  File "/usr/lib/python3.6/site-packages/nova/conductor/manager.py", line 1379, in schedule_and_build_instances\n    instance_uuids, return_alternates=True)\n  File "/usr/lib/python3.6/site-packages/nova/conductor/manager.py", line 839, in _schedule_instances\n    return_alternates=return_alternates)\n  File "/usr/lib/python3.6/site-packages/nova/scheduler/client/query.py", line 42, in select_destinations\n    instance_uuids, return_objects, return_alternates)\n  File "/usr/lib/python3.6/site-packages/nova/scheduler/rpcapi.py", line 160, in select_destinations\n    return cctxt.call(ctxt, \'select_destinations\', **msg_args)\n  File "/usr/lib/python3.6/site-packages/oslo_messaging/rpc/client.py", line 181, in call\n    transport_options=self.transport_options)\n  File "/usr/lib/python3.6/site-packages/oslo_messaging/transport.py", line 129, in _send\n    transport_options=transport_options)\n  File "/usr/lib/python3.6/site-packages/oslo_messaging/_drivers/amqpdriver.py", line 674, in send\n    transport_options=transport_options)\n  File "/usr/lib/python3.6/site-packages/oslo_messaging/_drivers/amqpdriver.py", line 664, in _send\n    raise result\nnova.exception_Remote.NoValidHost_Remote: No valid host was found. \nTraceback (most recent call last):\n\n  File "/usr/lib/python3.6/site-packages/oslo_messaging/rpc/server.py", line 235, in inner\n    return func(*args, **kwargs)\n\n  File "/usr/lib/python3.6/site-packages/nova/scheduler/manager.py", line 199, in select_destinations\n    raise exception.NoValidHost(reason="")\n\nnova.exception.NoValidHost: No valid host was found. \n\n'} |
```

**Comment by pradiptapks** [comment]
Looks like a misconfiguration which leads to the `No valid host was found` error.

**Comment by kemmy22** [comment]
what was the node type for controller nodes? was it a mixed env where in the nodes which were of type controller were also of type compute or it was a different node type for controllers?

**Comment by sanjaychari** [comment]
This environment has 5 1029u systems and 12 1029p systems. Ideally, Jetpack should adjust the compute count automatically (when compute count is not explicitly mentioned in group_vars/all.yml) when a baremetal node fails introspection and gets deleted, irrespective of node type as tagging happens after introspection, but it didn't. Compute count was not decreased and the deployment failed because of that, because there were only 2 baremetal nodes available but OSP was trying to deploy 3 controllers.

**Comment by asyedham** [comment]
Fixed #491
